sdk/nodejs/cmd/dynamic-provider/index.ts

```diff
@@ -57,9 +57,34 @@ process.on("exit", (code: number) => {
     }
 });
 
+const providerCache: { [key: string]: dynamic.ResourceProvider} = {};
+
+const hashCode = (str: string) => {
+    let hash = 0;
+    let i: number;
+    let chr: number;
+    if (str.length === 0) return hash;
+    for (i = 0; i < str.length; i++) {
+        chr = str.charCodeAt(i);
+        hash = ((hash << 5) - hash) + chr;
+        hash |= 0; // Convert to 32bit integer
+    }
+    return hash;
+};
```

Comment:
I think we should just use the provider string itself as the cache key. Using the hash code without keeping the original string around could lead to incorrect behavior if two different strings hash to the same value.

Reply:
Good point, at 1k providers the chance of collision becomes 1/10000. Storing the entire provider is probably not a big deal as the common case is multiple instances of a few providers over thousands of single instances of unique providers.

Reply:
Done.